util/internal/test/common/src/main/java/org/hibernate/search/util/impl/test/extension/parameterized/ParameterizedClassExtension.java

```diff
@@ -199,7 +216,10 @@ public void interceptTestTemplateMethod(Invocation<Void> invocation,
 					read( extensionContext, StoreKey.TEST_TO_RUN, ParameterizedTestMethodInvoker.class );
 			testMethod.invoke( extensionContext.getRequiredTestInstance() );
 			envInitialized = !envInitializationFailed;
+
+			if ( !hasMoreTests ) {
+				cleanUp.accept( ExtensionScope.PARAMETERIZED_CLASS_SETUP );
+			}
```

Comment:
I've tried a few things but because of this:

> That means that, given two extensions Extension1 and Extension2 with Extension1 registered before Extension2, any "before" callbacks implemented by Extension1 are guaranteed to execute before any "before" callbacks implemented by Extension2. Similarly, given the two same two extensions registered in the same order, any "after" callbacks implemented by Extension1 are guaranteed to execute after any "after" callbacks implemented by Extension2. Extension1 is therefore said to wrap Extension2.

the only way I figured that worked is to register an extension with the `@ParameterizedClass` annotation (see the `ParameterizedClassCleanupExtension`). It means that we need to store the boolean in the extension store, and read it in the other extension, but ... it works 😃